### ISSUE TITLE
Chatty log output

### ISSUE THREAD

**Issue by squarefrog** [robertklep/homebridge-nefit-easy#21]
Firstly, thanks for the plugin! Its really handy being able to ask Siri to turn our heating on.

I wonder if you'd be open to a configuration option to reduce the chattiness of the logging? I went to look at my log for another plugin yesterday, and it took a long time to find it as the log was filled with messages like below:

```
[9/30/2019, 4:49:04 PM] [Thermostat] Getting current temperature...
[9/30/2019, 4:49:05 PM] [Thermostat] ...target temperature is 21
[9/30/2019, 4:49:05 PM] [Thermostat] ...current temperature is 21.1
[9/30/2019, 4:49:05 PM] [Thermostat] ...current state is off
[9/30/2019, 4:50:00 PM] [Thermostat] Getting current state..
[9/30/2019, 4:50:00 PM] [Thermostat] Getting target temperature...
[9/30/2019, 4:50:00 PM] [Thermostat] Getting current temperature...
[9/30/2019, 4:50:01 PM] [Thermostat] ...target temperature is 21
[9/30/2019, 4:50:01 PM] [Thermostat] ...current temperature is 21.1
[9/30/2019, 4:50:01 PM] [Thermostat] ...current state is off
[9/30/2019, 4:50:01 PM] [Thermostat] Getting current state..
[9/30/2019, 4:50:01 PM] [Thermostat] Getting target temperature...
[9/30/2019, 4:50:01 PM] [Thermostat] Getting current temperature...
[9/30/2019, 4:50:02 PM] [Thermostat] ...target temperature is 21
[9/30/2019, 4:50:02 PM] [Thermostat] ...current temperature is 21.1
[9/30/2019, 4:50:02 PM] [Thermostat] ...current state is off
```

I think a possible approach would be to have a config option in `config.json`, something like: 

```
"accessories": [
    ...
    {
        "accessory" : "NefitEasy",
        "name"      : "thermostaat",
        "options"   : {
            "serialNumber" : "NEFIT_SERIAL_NUMBER",
            "accessKey"    : "NEFIT_ACCESS_KEY",
            "password"     : "NEFIT_PASSWORD",
            "logging"      : false
        }
    }
]
```

This would then keep the default verbose logging behaviour, while allowing the rest of us to opt out if needed. Alternatively, my preferred option, would be to disable the vebose logging, and have it opt-in, but I can understand if you'd be unwilling to change this behaviour.

Let me know if you'd be open to a Pull Request, as I'll probably add this functionality for myself in a fork.

**Comment by squarefrog** [comment]
On reflection I think this could be improved with a 3 tier logging option:

### Verbose (default)

Log all get and set requests. 

### Info

Log only set requests

### None

No log statements, only thrown errors. Not sure this one is useful.

**Comment by robertklep** [comment]
@squarefrog certainly open to a PR for this, I agree that the chattiness can be reduced.

Note that the `NefitEasyAccessory` class uses the `log` function passed to its constructor by Homebridge, so Homebridge already has a logging infrastructure in place. I'm not sure how it works (never looked at it), but it would be great if it were possible to base your changes on that.

**Comment by squarefrog** [comment]
@robertklep Good idea looking into the `log` function. Looking at [the source](https://github.com/nfarina/homebridge/blob/0d77bb93d33a7b6e158efe4b4d546636d976d5c7/lib/logger.js#L44-L59), we may be able to use these by making the get requests `log.debug`, and the rest `log.info`. I will take a look at these tonight.

**Comment by robertklep** [comment]
@squarefrog or perhaps use the [`log`](https://github.com/nfarina/homebridge/blob/0d77bb93d33a7b6e158efe4b4d546636d976d5c7/lib/logger.js#L61) method, which seems to accepts a log level. That way, you only have to read and pass the log level from the config to the calls to `this.log()` (at least, I think).

**Comment by squarefrog** [comment]
Yes, thats how I understand it. Looking through homebridge's source to see if this would work

**Comment by robertklep** [comment]
Just published `homebridge-nefit-easy@2.2.0` to the NPM repo.
